drivers/pwm/pwm_nrf5_sw.c

```diff
@@ -15,17 +15,30 @@
 #include <logging/log.h>
 LOG_MODULE_REGISTER(pwm_nrf5_sw);
 
+#define GENERATOR_NODE DT_PHANDLE(DT_DRV_INST(0), generator)
+
+#if DT_NODE_HAS_COMPAT(GENERATOR_NODE, nordic_nrf_rtc)
+#define USE_RTC		1
+#define GENERATOR_ADDR	((NRF_RTC_Type *) DT_REG_ADDR(GENERATOR_NODE))
+BUILD_ASSERT(DT_INST_PROP(0, clock_prescaler) == 0,
+	     "Only clock-prescaler = <0> is supported when used with RTC");
+#else
+#define USE_RTC		0
+#define GENERATOR_ADDR	((NRF_TIMER_Type *) DT_REG_ADDR(GENERATOR_NODE))
+#endif
+
 /* One compare channel is needed to set the PWM period, hence +1. */
-#if ((DT_INST_PROP(0, channel_count) + 1) > \
-	(_CONCAT( \
-		_CONCAT(TIMER, DT_INST_PROP(0, timer_instance)), \
-		_CC_NUM)))
+#if ((DT_INST_PROP(0, channel_count) + 1) > DT_PROP(GENERATOR_NODE, cc_num))
 #error "Invalid number of PWM channels configured."
 #endif
 #define PWM_0_MAP_SIZE DT_INST_PROP(0, channel_count)
 
 struct pwm_config {
+#if USE_RTC
+	NRF_RTC_Type *rtc;
+#else
 	NRF_TIMER_Type *timer;
+#endif
```

Comment:
How about using a union here? Maybe this could please the compliance check?

Reply:
Indeed, using `union` helped with `checkpatch.pl`. `#ifdef` might have confused it somehow.